src/main/java/core/basesyntax/service/impl/WriterImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.service.Writer;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class WriterImpl implements Writer {
+    private static final String REPORT_HEAD_LINE = "fruit,quantity";
+    private static final char COMMA = ',';
+    private StorageDao storageDao;
+
+    public WriterImpl(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void createReport(String reportPath) {
+        File report = new File(reportPath);
```

Comment:
this violates SRP - separate report generation from writing that report to a file, writer should be agnostic to input data, just writes what it's given